File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.py
class Solution:
    def longestBeautifulSubstring(self, word: str) -> int:
        if len(word) < 5:
            return 0
        ans = 0
        a = e = i = o = u = False
        idx = 0
        while idx < len(word):
            s = idx
            while idx < len(word) and word[idx] == 'a':
                idx += 1
                a = True
            while idx < len(word) and a and  word[idx] == 'e':
                idx += 1
                e = True
            while idx < len(word) and e and word[idx] == 'i':
                idx += 1
                i = True
            while idx < len(word) and i and word[idx] == 'o':
                idx += 1
                o = True
            while idx < len(word) and o and word[idx] == 'u':
                idx += 1
                u = True
            if u:
                ans = max(ans, idx-s)
            a = e = i = o = u = False
            if idx == s:
                idx += 1
        return ans

File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.java
class Solution {
    public int longestBeautifulSubstring(String word) {
        boolean a = false, e = false, i = false, o = false, u = false;
        int ans = 0, idx = 0;
        
        while(idx < word.length()){
            int start = idx;
            while(idx < word.length() && word.charAt(idx) == 'a'){
                idx++;
                a = true;
            }
            while(idx < word.length() && a && word.charAt(idx) == 'e'){
                idx++;
                e = true;
            }
            while(idx < word.length() && e && word.charAt(idx) == 'i'){
                idx++;
                i = true;
            }
            while(idx < word.length() && i && word.charAt(idx) == 'o'){
                idx++;
                o = true;
            }
            while(idx < word.length() && o && word.charAt(idx) == 'u'){
                idx++;
                u = true;
            }
            if(u){
                ans = Math.max(ans, idx-start);
            }
            a = e = i = o = u = false;
            if(start == idx)
                idx++;
        }
        return ans;
    }
}